Frameworks/Foundation/NSDictionary.mm

```diff
@@ -99,8 +99,11 @@ - (NSArray*)allObjects {
         return nil;
     }
 
-    StrongId<NSArray> keyArray = [_keyEnumerator allObjects];
-    StrongId<NSMutableArray> toRet = [NSMutableArray new];
+    StrongId<NSArray> keyArray;
+    keyArray.attach([_keyEnumerator allObjects]);
+
+    StrongId<NSMutableArray> toRet;
```

Comment:
true that would be autoreleased but you are adding complication to the object lifetime by doing an autorelease on copy rule object. There is no need to autorelease and its not necessarily clear from the callee perspective when an autorelease will actually take place which makes the object lifetime longer than needed and may in fact cause some issues with resource de-init (see Java's or other garabage collected language's issues around closing files deterministically etc.)

---
In reply to: [73973850](https://github.com/Microsoft/WinObjC/pull/851#discussion_r73973850) [](ancestors = 73973850,73973752,73973293)